src/gui/executionlogwidget.cpp

```diff
@@ -32,75 +32,45 @@
 #include <QPalette>
 
 #include "base/global.h"
-#include "loglistwidget.h"
+#include "base/logger.h"
+#include "log/logfiltermodel.h"
+#include "log/loglistview.h"
+#include "log/logmodel.h"
 #include "ui_executionlogwidget.h"
 #include "uithememanager.h"
 
-ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes &types)
+ExecutionLogWidget::ExecutionLogWidget(QWidget *parent, const Log::MsgTypes types)
     : QWidget(parent)
     , m_ui(new Ui::ExecutionLogWidget)
-    , m_msgList(new LogListWidget(MAX_LOG_MESSAGES, types, this))
-    , m_peerList(new LogListWidget(MAX_LOG_MESSAGES, Log::ALL, this))
+    , m_msgFilterModel(new LogFilterModel(types, this))
 {
     m_ui->setupUi(this);
 
+    LogListView *msgList = new LogListView(this);
+    LogModel *msgModel = new LogModel(this);
+    m_msgFilterModel->setSourceModel(msgModel);
+    msgList->setModel(m_msgFilterModel);
+    connect(msgList, &LogListView::resetModel, [msgModel]() { msgModel->reset(); });
+
+    LogListView *peerList = new LogListView(this);
+    LogPeerModel *peerModel = new LogPeerModel(this);
+    peerList->setModel(peerModel);
+    connect(peerList, &LogListView::resetModel, [peerModel]() { peerModel->reset(); });
+
 #ifndef Q_OS_MACOS
     m_ui->tabConsole->setTabIcon(0, UIThemeManager::instance()->getIcon("view-calendar-journal"));
     m_ui->tabConsole->setTabIcon(1, UIThemeManager::instance()->getIcon("view-filter"));
 #endif
-    m_ui->tabGeneral->layout()->addWidget(m_msgList);
-    m_ui->tabBan->layout()->addWidget(m_peerList);
-
-    const Logger *const logger = Logger::instance();
-    for (const Log::Msg &msg : asConst(logger->getMessages()))
-        addLogMessage(msg);
-    for (const Log::Peer &peer : asConst(logger->getPeers()))
-        addPeerMessage(peer);
-    connect(logger, &Logger::newLogMessage, this, &ExecutionLogWidget::addLogMessage);
-    connect(logger, &Logger::newLogPeer, this, &ExecutionLogWidget::addPeerMessage);
+    m_ui->tabGeneral->layout()->addWidget(msgList);
+    m_ui->tabBan->layout()->addWidget(peerList);
 }
 
 ExecutionLogWidget::~ExecutionLogWidget()
 {
     delete m_ui;
 }
 
-void ExecutionLogWidget::showMsgTypes(const Log::MsgTypes &types)
-{
-    m_msgList->showMsgTypes(types);
-}
-
-void ExecutionLogWidget::addLogMessage(const Log::Msg &msg)
+void ExecutionLogWidget::showMsgTypes(const Log::MsgTypes types)
```

Comment:
Should this be renamed to `setMsgTypes()`?

Reply:
done.